MonumentAddons.cs

```diff
@@ -3631,6 +3607,249 @@ private interface IDeepCollection
+                    }
+                }
+
+                return phoneName;
+            }
+
+            public static string GetFTLTrainStationPhoneName(MonumentInfo attachedMonument, string tunnelName, Vector3 position, MonumentHelper monumentHelper) 
+            {
+                var phoneName = string.IsNullOrEmpty(attachedMonument.displayPhrase.translated) 
+                    ? $"{Tunnel} {tunnelName}" 
+                    : $"{Tunnel} {attachedMonument.displayPhrase.translated} {tunnelName}";
+
+                var shortname = GetShortName(attachedMonument.name);
+
+                if (ShouldAppendCoordinate(shortname, monumentHelper))
+                    phoneName += $" {PhoneController.PositionToGridCoord(position)}";
+
+                return phoneName;
+            }
+
+            public static string GetUnderwaterLabPhoneName(DungeonBaseLink link, Vector3 position)
+            {
+                var floors = link.Dungeon.Floors;
+                var gridCoordinate = PhoneController.PositionToGridCoord(position);
+
+                for (int i = 0; i < floors.Count; i++)
+                {
+                    if (floors[i].Links.Contains(link))
+                    {
+                        var roomLevel = $"L{1 + i}";
+                        var roomType = link.Type.ToString();
+                        var roomNumber = 1 + floors[i].Links.IndexOf(link);
+                        var roomName = $"{roomLevel} {roomType} {roomNumber}";
+
+                        return $"{UnderwaterLab} {gridCoordinate} {roomName}";
+                    }
+                }
+
+                return $"{UnderwaterLab} {gridCoordinate}";
+            }
+
+            public static string GetDynamicMonumentPhoneName(DynamicMonument monument, Telephone phone)
+            {
+                if (monument.RootEntity is CargoShip)
+                {
+                    return $"{CargoShip} {monument.EntityId}";
+                }
+                
+                return $"{phone.GetDisplayName()} {monument.EntityId}";
+            }
+
+            public static bool ShouldAppendCoordinate(string monumentShortName, MonumentHelper monumentHelper)
+            {
+                string[] monumentVariants = 
+                {
+                    "fishing_village_b",
+                    "fishing_village_c",
+                    "harbor_1",
+                    "harbor_2",
+                    "power_sub_big_1",
+                    "power_sub_big_2",
+                    "power_sub_small_1",
+                    "power_sub_small_2",
+                    "water_well_a",
+                    "water_well_b",
+                    "water_well_c",
+                    "water_well_d",
+                    "water_well_e",
+                    "entrance_bunker_a",
+                    "entrance_bunker_b",
+                    "entrance_bunker_c",
+                    "entrance_bunker_d",
+                };
```

Comment:
An array is fine, but we shouldn't construct it every time this method is called. Instead, make it a `static readonly` field of the class, and give it a more descriptive name.